host/xtest/xtest_main.c

```diff
@@ -147,27 +160,51 @@ int main(int argc, char *argv[])
 	else if (argc > 1 && !strcmp(argv[1], "--stats"))
 		return stats_runner_cmd_parser(argc - 1, &argv[1]);
 
-	while ((opt = getopt(argc, argv, "d:l:t:h")) != -1)
+	while ((opt = getopt(argc, argv, "d:l:t:h:k")) != -1) {
 		switch (opt) {
 		case 'd':
 			_device = optarg;
+			last_gen_option = optind;
 			break;
 		case 'l':
 			p = optarg;
+			last_gen_option = optind;
 			break;
 		case 't':
 			test_suite = optarg;
+			last_gen_option = optind;
 			break;
 		case 'h':
 			usage(argv[0]);
 			return 0;
+		case '?':
+			if (optopt == 'x') {
+				/*
+				 * -w and -x are not an option processed here,
```

Comment:
"The -x option is not processed here, it is part of the test IDs"

Reply:
ok